tuning/sft_trainer.py

```diff
@@ -693,6 +687,33 @@ def main():
             )
             sys.exit(INTERNAL_ERROR_EXIT_CODE)
 
+    if isinstance(tune_config, peft_config.LoraConfig):
+        try:
+            if training_args.save_model_dir:
+                # Write number of added tokens to artifacts
+                with open(
+                    os.path.join(
+                        training_args.save_model_dir, "added_tokens_info.json"
+                    ),
+                    "w",
+                    encoding="utf-8",
+                ) as f:
+                    json.dump(additional_train_info["added_tokens_info"], f)
```

Comment:
no , writing it in all cases to differentiate between "tuning was not done on latest code / no tokens were added". The file will always exist, it will contain value 0 if no tokens added

Reply:
added a check in script to exit early if num_added_tokens == 0 though https://github.com/foundation-model-stack/fms-hf-tuning/pull/338/files#diff-443a0a5c8690e1a9251b2805f8e5850c97381d40b47cd90c5716eeaca84706d9R61